### PR TITLE
smartgithg: 17.1.4 -> 18.1.4

### PR DESCRIPTION
**Pull request by baracoder**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by fgaz** [comment]
@GrahamcOfBorg build

File: pkgs/applications/version-management/smartgithg/default.nix [hunk 1]
```diff
@@ -37,6 +37,8 @@ stdenv.mkDerivation rec {
     mkdir -pv ${pkg_path}
     # unpacking should have produced a dir named 'smartgit'
     cp -a smartgit/* ${pkg_path}
+    # prevent using packaged jre
+    mv ${pkg_path}/jre/bin/java ${pkg_path}/jre/bin/java-unused
```
Comment (Mic92) [inline pkgs/applications/version-management/smartgithg/default.nix]: Would removing the whole jre work or just symlinking our jre inplace to that path?

Comment (baracoder) [reply pkgs/applications/version-management/smartgithg/default.nix]: Yes, the whole `jre/` directory can be removed. I will update it.

**Comment by xeji** [comment]
unfree package, build checked locally

**Pull request merged**